### ISSUE TITLE
move_and_slide fails when KinematicBody is in origin

### ISSUE THREAD

**Issue by pseidemann** [godotengine/godot#59826]
### Godot version

3.4.4.stable

### System information

macOS 12.1 (21C52)

### Issue description

As the floor I have a StaticBody containing a MeshInstance (plane) and a CollisionShape (box). The collision shape is translated -1 on y so that it matches with the x/z plane.
The player is a KinematicBody which stands on this floor. All nodes being in the x/z origin.
And I have a GridMap with at least one cell filled.

the KinematicBody has the following code:

```gdscript
extends KinematicBody

var velocity = Vector3()

func _physics_process(delta):
	velocity.y -= 9.8 * delta
	velocity = move_and_slide(velocity, Vector3.UP)
	print(is_on_floor())
```
is_on_floor() is false even though the player is on a floor. Also there is sometime jitter visible.
The player gets "stuck" on the ground and I get the following error logged:

```
E 0:00:00.452   recover_from_penetration: Condition "shape_idx < 0 || shape_idx >= cs->getNumChildShapes()" is true. Returned: false
  <C++ Source>  modules/bullet/space_bullet.cpp:1306 @ recover_from_penetration()
  <Stack Trace> Player.gd:35 @ _physics_process()
```

When I move out the x/z origin on the plane, the error goes away. When I move back to the origin (doesn't have to be exact), the error starts again and the player is again stuck. Another fix is to move the collision shape box of the floor above the x/z plane. But once it's again below the x/z plane, the error happens again.

### Steps to reproduce

Just start the project.

### Minimal reproduction project

[collision_bug.zip](https://github.com/godotengine/godot/files/8404018/collision_bug.zip)

**Comment by pseidemann** [comment]
I was now able to find out what caused the bug in my game: it's a GridMap with needs at least one cell filled. Now I could create a minimal reproduction.
[collision_bug.zip](https://github.com/godotengine/godot/files/8404018/collision_bug.zip)

**Comment by Riteo** [comment]
@pseidemann does this happen only with such a gridmap or also with similar static bodies?

**Comment by pseidemann** [comment]
the gridmap is not the static body here but it triggers the bug somehow. I couldn't reproduce it when I remove the gridmap

**Comment by akien-mga** [comment]
Fixed by #59864.
